core/src/main/scala/ai/lum/odinson/state/EventPromoter.scala

```diff
@@ -40,13 +41,62 @@ class EventPromoter {
     )
   }
 
+  protected def promoteOdinsonMatch(odinsonMatch: OdinsonMatch, odinsonScoreDoc: OdinsonScoreDoc): Array[LabeledNamedOdinResults]
+
+  protected def promoteOdinResults(odinResults: OdinResults): Array[LabeledNamedOdinResults] = {
+    odinResults.scoreDocs.flatMap { odinsonScoreDoc =>
+      odinsonScoreDoc.matches.flatMap { odinsonMatch =>
+        promoteOdinsonMatch(odinsonMatch, odinsonScoreDoc)
+      }
+    }
+  }
+}
+
+class FlatEventPromoter extends EventPromoter {
+
+  protected def promoteOdinsonMatch(odinsonMatch: OdinsonMatch, odinsonScoreDoc: OdinsonScoreDoc): Array[LabeledNamedOdinResults] = {
+    val result = odinsonMatch match {
+      case eventMatch: EventMatch =>
+        // Turn this back on for paranoia mode.
+        // {
+        //  val namesAndArgumentMetadatas = eventMatch.argumentMetadata.groupBy(_.name)
+        //  namesAndArgumentMetadatas.foreach { case (name, argumentMetadatas) =>
+        //    // Those with the same name should all be the same.
+        //    argumentMetadatas.distinct.length == 1
+        //  }
+        // }
+        val namesToPromote: Set[String] = eventMatch.argumentMetadata
+            .map(_.name)
```

Comment:
correct me if I'm wrong, but don't we need to check the `.promote` attribute here?